.gitattributes

```diff
@@ -0,0 +1,65 @@
+###############################
+# Git Line Endings            #
+###############################
+
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto eol=lf
+
+# Explicitly declare text files you want to always be normalized and converted
+# to native line endings on checkout.
+*.txt text
+*.yml text
+*.py text
+*.cmake text
+*.c text
+*.cpp text
+*.h text
+*.hpp text
+.gitattributes text
+.gitignore text
+.dockerignore text
+*.md text
+.clang-format text
+.clang-tidy text
+Dockerfile text
+*.Dockerfile text
+Dockerfile.* text
+LICENSE text
+
+# Force batch scripts to always use CRLF line endings so that if a repo is accessed
+# in Windows via a file share from Linux, the scripts will work.
+*.{cmd,[cC][mM][dD]} text eol=crlf
+*.{bat,[bB][aA][tT]} text eol=crlf
+
+# Force bash scripts to always use LF line endings so that if a repo is accessed
+# in Unix via a file share from Windows, the scripts will work.
+*.sh text eol=lf
+
+###############################
+# Git Large File System (LFS) #
```

Comment:
Git LFS is not free, and there is a size limit on GitHub. Not sure if we should use it by default for all of these extensions.

Reply:
https://github.com/git-lfs/git-lfs/blob/main/LICENSE.md

Reply:
> https://github.com/git-lfs/git-lfs/blob/main/LICENSE.md

https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage#about-billing-for-git-large-file-storage

1GB is free

Reply:
Yeah, the software is free but the storage is not.

Reply:
Sorry, I am used to self-hosted solutions. Well if you have big binary assets for your application you cannot let them getting diffed  like a text it is a big git slowdown with out any meaningful outcome.

Reply:
Given that users will upload only assets that are necessary for their applications and not binaries the free 1G should be more than enough. Also there is the option to package and version your assets using conan, but this is a very specific use case, (probably for an other MR) However, implementing something like this I am afraid push too far for a multi purpose template project.

Reply:
I still want this to be reverted. You can add a link to a Gist explaining this instead. Repos with Git LFS are also slower to clone

Reply:
OK I  removed that part of the .gitattributes. I still believe that setting lfs from the early stages of the project to decrease the overall repo size on the client worth the trade off.

Reply:
https://stackoverflow.com/questions/35575400/what-is-the-advantage-of-git-lfs

Reply:
You could leave it there but just commented out. If someone has very big files, it is really useful to use LFS.

I totally agree that LFS is useful, and solves one of the issues of Git. My main problem with it is that it is not free on GitHub. 
I used to use it on GitHub for free, and I constantly hit the limit. The limit prevented any clones or pushes! That's why I prefer this to be opt-in rather than by default.